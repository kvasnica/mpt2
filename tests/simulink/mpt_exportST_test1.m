function mpt_exportST_test1

% tests the mpt_exportST() function which should generate a C-code which
% implements binary search trees:
% http://control.ee.ethz.ch/~mpt/hg/mpt-stm?cs=e8292128438b

%fprintf('no search tree for this controller, the function should fail\n');
load ctrl1
lasterr('');
try
    mpt_exportST(ctrl);
    worked = 1;
catch
    worked = 0;
end
mbg_assertfalse(worked);
mbg_assertcontains(lasterr, 'Search tree is not present in the controller. Use "mpt_searchTree(ctrl)" to compute it.');


fprintf('\nSearch tree should be written to the default file "searchtree.c"\n');
load ctrl_1d_searchtree
mpt_exportST(ctrl_st);
fileisthere = exist('searchtree.c', 'file');
if fileisthere,
    % remove the file
    try
        delete('searchtree.c');
    end
end
mbg_asserttrue(fileisthere);

fprintf('\nSearch tree should be written to a custom file "myctrl.c"\n');
load ctrl_1d_searchtree
mpt_exportST(ctrl_st, 'myctrl.c');
fileisthere = exist('myctrl.c', 'file');
if fileisthere,
    % remove the file
    try
        delete('myctrl.c');
    end
end
mbg_asserttrue(fileisthere);
