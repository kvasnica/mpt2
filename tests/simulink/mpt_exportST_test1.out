
Search tree should be written to the default file "searchtree.c"
Output written to "searchtree.c".

Search tree should be written to a custom file "myctrl.c"
Output written to "myctrl.c".
